imagescan/delta.go

```diff
@@ -203,11 +207,18 @@ func (d *deltaState) upsertImages(pod *corev1.Pod) {
 
 		key := cs.ImageID
 		nodeName := pod.Spec.NodeName
+		arch := "amd64"
+		n, ok := d.nodes[nodeName]
+		if ok {
+			arch = n.architecture
+			key += arch
```

Comment:
How did you decide on the key format?

Reply:
because image id is the same for all the architectures (think fat digest), we need to perform scan on images for all different architectures. Without this cache we would scan all of the images multiple times (every time informer reports that pod changed). ImageID+architecture as key allows us to scan same imageID for different architecture